tensorflow/core/ops/math_ops.cc

```diff
@@ -478,6 +478,14 @@ REGISTER_OP("Ceil")
 Returns element-wise smallest integer in not less than x.
 )doc");
 
+REGISTER_OP("Rint")
+    .Input("x: T")
+    .Output("y: T")
+    .Attr("T: {float, double}")
+    .Doc(R"doc(
+Returns element-wise integer closest to x.
```

Comment:
@geoffreyi would have useful knowledge here.

On Wed, Aug 31, 2016 at 10:41 AM, Fabrizio Milo <notifications@github.com>
wrote:

> In tensorflow/core/ops/math_ops.cc
> <https://github.com/tensorflow/tensorflow/pull/4113#discussion_r77035381>:
>
> > @@ -478,6 +478,14 @@ REGISTER_OP("Ceil")
> >  Returns element-wise smallest integer in not less than x.
> >  )doc");
> >
> > +REGISTER_OP("Rint")
> > +    .Input("x: T")
> > +    .Output("y: T")
> > +    .Attr("T: {float, double}")
> > +    .Doc(R"doc(
> > +Returns element-wise integer closest to x.
>
> Interesting the underyling std::rint operator behaves differently based
> on the current rounding mode: std::fesetround
> <http://en.cppreference.com/w/cpp/numeric/fenv/FE_round>. Seems TF does
> not set this explicitly in the code base neither does Eigen (checked this
> doing an ack fsetround). Should we set the mode explicitly ?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/tensorflow/tensorflow/pull/4113/files/0bfab954645d0a2c02f3e2f54382bdd84f950024#r77035381>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAjO_QKStgVDqNctouxhPggzc7BXFyKoks5qlbzZgaJpZM4JxJ4z>
> .
>